packages/core/src/deduplication.ts

```diff
@@ -0,0 +1,99 @@
+import { Patient, Reference, Resource } from '@medplum/fhirtypes';
+import { MedplumClient } from './client';
+import { createReference, deepClone } from './utils';
+
+interface MergedPatients {
+  readonly src: Patient;
+  readonly target: Patient;
+}
+
+interface Subject {
+  subject?: Reference<Patient>;
+}
+
+type ResourceWithSubject = Resource & Subject;
+
+export function linkPatientRecords(src: Patient, target: Patient): MergedPatients {
+  const targetLinks = target.link ?? [];
+  targetLinks.push({ other: createReference(src), type: 'replaces' });
+  const targetCopy = deepClone(target);
```

Comment:
On the previous line, you already modify the incoming object by pushing to target.link. 
If the contract is that we clone the resource, then we should copy as at teh first step